### PR TITLE
in sample update handler, add sample if it's not already being tracked

### PR DESCRIPTION
**Pull request by shriramshankar**

### CODE REVIEW

File: MultiTable/src/RealtimeChangeHandler.js [hunk 1]
```diff
@@ -58,6 +58,9 @@ function onSampleUpdate(data, change) {
         if (index >= 0) {
           subject.samples[index] = change;
         }
+        else {
```
Comment (iamigo) [inline MultiTable/src/RealtimeChangeHandler.js]: `} else {`

File: MultiTable/test/RealtimeChangeHandler.js [hunk 2]
```diff
@@ -14,11 +14,74 @@ describe('./test/RealtimeChangeHandler.js >', () => {
...
     const chg = { 'sample.add': { name: 'a|b', aspect: { name: 'b'} } };
     RealtimeChangeHandler.handle(chg, data);
-    expect(data instanceof SubjectGroups).to.equal.true;
+    expect(data instanceof SubjectGroups).to.be.true;
     expect(data).to.have.property('rootSubject', 'a');
     expect(data.map).to.have.property('a');
   });
 
+  describe('handle: updateSample >', () => {
+
```
Comment (iamigo) [inline MultiTable/test/RealtimeChangeHandler.js]: no empty line here

**Comment by iamigo** [comment]
To merge, will also need zip file in the PR

**Comment by shriramshankar** [comment]
Yes, I will create the zip, test it in the sandbox and then upload it here.

File: MultiTable/lens.json [hunk 3]
```diff
@@ -37,6 +37,7 @@
     "1.0.19": "Aspect name case insensitivity - missed one",
     "1.0.20": "Add google analytics tracking for search, onclick modals and links, and toggle checkbox",
     "1.0.21": "Prevent reverse tabnabbing",
-    "1.0.22": "Add RealtimeChangeHandler arg validation and tests"
+    "1.0.22": "Add RealtimeChangeHandler arg validation and tests",
+    "1.0.23": "Update onSampleUpdate to add a sample when not found"
```
Comment (jgraff2) [inline MultiTable/lens.json]: You also need to update the version up above.

File: MultiTable/test/RealtimeChangeHandler.js [hunk 4]
```diff
@@ -14,11 +14,73 @@ describe('./test/RealtimeChangeHandler.js >', () => {
...
+       * make sure that the subjects instance object of the subjectGroup has
+       * the updated sample
+       */
+      expect(data.map.fellowship.subjects.fellowship.samples)
+        .to.have.lengthOf(1);
+      expect(data.map.fellowship.subjects.fellowship.samples[0]).to
+        .eql(updatedSample);
+     });
+
+    it('updateSample should add sample to the subject array if not ' +
```
Comment (jgraff2) [inline MultiTable/test/RealtimeChangeHandler.js]: Would be good to also have a test that replicates the exact situation, where a subject is deleted then added back.

File: MultiTable/lens.json (hunk 3 above)
Comment (shriramshankar) [reply MultiTable/lens.json]: I do not understand what you are trying to say.

Comment (jgraff2) [reply MultiTable/lens.json]: line 7 of this file.

Comment (shriramshankar) [reply MultiTable/lens.json]: Got it.

Comment (iamigo) [reply MultiTable/lens.json]: maybe describe like this: "Update the sample update handler to add a sample if it's not already being tracked" ?

**Comment by shriramshankar** [comment]
@iamigo @jgraff2 PR updated with the version changes.

File: MultiTable/test/RealtimeChangeHandler.js (hunk 4 above)
Comment (shriramshankar) [reply MultiTable/test/RealtimeChangeHandler.js]: From the tests done today, it was clear that this fix does not solve the problem. I was able to narrow down the problem to the addSubject method and I am creating a bug describing the exact scenario on which the pods disappear. We can write the test as a part of that bug.

**Pull request merged**